### PR TITLE
i/builtin: Added RSA DS100 authenticator

### PR DESCRIPTION
**Pull request by GregoryRG**
Added RSA DS100 authenticator to the whitelist. RSA Security is the vendor and the authenticator is certified by the FIDO Alliance (certificate no. FIDO20020230501001).

### CODE REVIEW

**Comment by GregoryRG** [comment]
The FIDO MDS Explorer is an online tool that shows FIDO authenticators that are registered with the FIDO Alliance. If you search for "RSA DS100", the DS100 is visible.
https://opotonniee.github.io/fido-mds-explorer/

File: interfaces/builtin/u2f_devices.go [hunk 1]
```diff
@@ -177,6 +177,11 @@ var u2fDevices = []u2fDevice{
 		VendorIDPattern:  "1370",
 		ProductIDPattern: "0911",
 	},
+	{
```
Comment (Meulengracht) [inline interfaces/builtin/u2f_devices.go:180]: this change needs to be addressed in the unit tests as well

Comment (GregoryRG) [reply interfaces/builtin/u2f_devices.go:180]: Are there unit tests for any of the other entries in the u2fDevices array that I'm touching? I was trying to find a unit test for another u2fDevice that I could use as a model, but I didn't see any. Thanks.

Comment (alexmurray) [reply interfaces/builtin/u2f_devices.go:180]: As you can see in https://github.com/snapcore/snapd/actions/runs/7407288712/job/20153342164#step:13:385 the existing test is now failing with this change due to the addition of an extra element to this array - so the test needs to be updated to expect this new number of elements.

**Review by alexmurray — COMMENTED** [review]
Can you provide any reference (e.g. upstream datasheet / documentation etc) to the vendorID and productID values which you have used here?

**Comment by GregoryRG** [comment]
My apologies. I'll fix the unit test. Thanks.

**Comment by GregoryRG** [comment]
The data sheet for the RSA DS100 can be found here:
https://www.rsa.com/wp-content/uploads/Datasheet_IDPlusDS100_RSA_1022_Web.pdf

Unfortunately the datasheet doesn't list the vid or pid.

The document https://www.usb.org/sites/default/files/usb_vids_080223.pdf shows the vid for RSA Security as decimal 5601, which is 0x15e1. This is the value that I've added to the device entry u2f_devices.go. I don't think our product ID is registered anywhere.

**Review by alexmurray — APPROVED** [review]
Thanks for the extra details @GregoryRG - LGTM!

**Review by Meulengracht — APPROVED** [review]
Thank you, lgtm

**Comment by GregoryRG** [comment]
Could someone kindly tell me if the failed checks above were caused by my code changes? Thanks very much.

**Comment by alexmurray** [comment]
Failing spread tests do not appear to be related to this change.

**Pull request merged**